ethabi/src/token/lenient.rs

```diff
@@ -40,17 +40,32 @@ impl Tokenizer for LenientTokenizer {
 			return result;
 		}
 
-		let uint = u32::from_str_radix(value, 10)?;
-		Ok(pad_u32(uint))
+		let uint = Uint::from_dec_str(value)?;
+		Ok(uint.into())
 	}
 
+	// We don't have a proper signed int 256-bit long type, so here we're cheating. We build a U256
+	// out of it and check that it's within the lower/upper bound of a hypothetical I256 type: half
+	// the `U256::max_value().
 	fn tokenize_int(value: &str) -> Result<[u8; 32], Error> {
 		let result = StrictTokenizer::tokenize_int(value);
 		if result.is_ok() {
 			return result;
 		}
 
-		let int = i32::from_str_radix(value, 10)?;
-		Ok(pad_i32(int))
+		let abs = Uint::from_dec_str(value.trim_start_matches('-'))?;
+		let max = Uint::max_value() / 2;
+		let int = if value.starts_with('-') {
+			if abs > max {
+				return Err(Error::Other("int256 parse error: Underflow".into()));
+			}
+			!abs + 1
+		} else {
+			if abs > max {
```

Comment:
the range for signed is `[-m, m)` where `m = unsigned::max_value() / 2`

```suggestion
			if abs >= max {
```